mountpoint-s3-client/tests/endpoint_config.rs

```diff
@@ -36,33 +35,126 @@ async fn run_test<F: FnOnce(&str) -> EndpointConfig>(f: F) {
     check_get_result(result, None, &body[..]).await;
 }
 
-#[test_case(AddressingStyle::Automatic)]
-#[test_case(AddressingStyle::Path)]
+#[test_case(AddressingStyle::Automatic, "test_default_addressing_style")]
+#[test_case(AddressingStyle::Path, "test_path_addressing_style")]
 #[tokio::test]
-async fn test_addressing_style_region(addressing_style: AddressingStyle) {
-    run_test(|region| EndpointConfig::new(region).addressing_style(addressing_style)).await;
+async fn test_addressing_style_region(addressing_style: AddressingStyle, prefix: &str) {
+    run_test(
+        |region| EndpointConfig::new(region).addressing_style(addressing_style),
+        &get_unique_test_prefix(prefix),
+        get_test_bucket(),
+    )
+    .await;
 }
 
 #[cfg(feature = "fips_tests")]
 #[tokio::test]
 async fn test_fips_mount_option() {
```

Comment:
pre-existing: why "mount_option" in `test_fips_mount_option`? This is a test of the S3 client, I'd just call this something like: `test_use_fips`, which is the `EndpointConfig` option under test.